tests/unit/Statement/StatementListTest.php

```diff
@@ -260,6 +260,36 @@ public function testAddStatementWithGuid() {
 		$this->assertEquals( new StatementList( $statement ), $list );
 	}
 
+	public function testRemoveStatementsWithGuid() {
```

Comment:
Please create dedicated test method for different behaviours.

http://c2.com/cgi/wiki?ArrangeActAssert

Doing non-assertion stuff after doing asserts is the thing to avoid.